apps/protocol-frontend/src/components/EditContributionForm.tsx

```diff
@@ -84,9 +80,77 @@ const EditContributionForm = ({
 
   const combinedDaoListOptions = [...new Set([...daoReset, ...daoListOptions])];
 
+  useEffect(() => {
+    setValue('name', contribution?.name);
+    setValue('details', contribution?.details);
+    setValue('proof', contribution?.proof);
+    setValue('engagementDate', new Date(contribution?.date_of_engagement));
+    setValue('activityType', contribution?.activity_type.name);
+    setValue(
+      'daoId',
+      contribution?.guilds[0]?.guild.id
+        ? contribution?.guilds[0]?.guild.id
+        : daoReset[0].value,
+    );
+  }, [contribution, daoReset, setValue]);
+
+  const toast = useToast();
+  const [, setIpfsUri] = useState('');
+  const [selectedFile, setSelectedFile] = useState<File | null>(null);
+  const fileInputField = useRef<HTMLInputElement>(null);
+  const [fileError, setFileError] = useState<string | null>(null);
+  const [, setIsUploading] = useState(false);
+
+  const handleFileUploadButtonClick = () => {
+    if (fileInputField.current !== null) {
+      fileInputField.current.click();
+    }
+  };
+
+  const handleImageSelect = async (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const file = event?.target.files ? event.target.files[0] : null;
+    if (file !== null) {
+      setValue('proof', '');
+      setFileError(null);
+      if (file.size > MAX_FILE_UPLOAD_SIZE) {
+        setFileError(
+          'File is too large. Please select something smaller than 5 MB.',
+        );
+        return;
+      }
+    }
+    if (file && fileError === null) {
+      setIsUploading(true);
+      setSelectedFile(file);
+      const ipfsImageUri = await uploadFileIpfs(file, true);
+      if (ipfsImageUri) {
+        setIpfsUri(ipfsImageUri);
+        setValue('proof', ipfsImageUri);
+        setIsUploading(false);
+      }
+    }
+  };
+
   const updateContributionHandler: SubmitHandler<
     ContributionFormValues
   > = async values => {
+    if (selectedFile && fileError === null) {
+      try {
+        await uploadFileIpfs(selectedFile, false);
+      } catch (error) {
+        console.error(error);
+        toast({
```

Comment:
Should this failure fail adding the contribution?

Reply:
That's a great question. I wonder if we'd want this to fail since it the user wouldn't be adding in their intended Contribution with a failed proof, but I also can see us allowing it since the proof is optional.

Reply:
@alexkeating new behavior: 

https://user-images.githubusercontent.com/9438776/192346457-38afea05-4e85-4ccb-912c-627148055959.mp4